src/Workspaces/Core/Portable/Storage/PersistentStorageExtensions.cs

```diff
@@ -23,16 +25,37 @@ public static IChecksummedPersistentStorageService GetPersistentStorageService(t
             switch (database)
             {
                 case StorageDatabase.SQLite:
-                    return services.GetService<ISQLiteStorageServiceFactory>()?.Create(configuration) ??
+                    return services.GetService<ISQLiteStorageServiceFactory>()?.Create() ??
                            NoOpPersistentStorageService.GetOrThrow(configuration);
 
                 case StorageDatabase.CloudCache:
-                    return services.GetService<ICloudCacheStorageServiceFactory>()?.Create(configuration) ??
+                    return services.GetService<ICloudCacheStorageServiceFactory>()?.Create() ??
                            NoOpPersistentStorageService.GetOrThrow(configuration);
 
                 default:
                     return NoOpPersistentStorageService.GetOrThrow(configuration);
             }
         }
+
+        public static ValueTask<IChecksummedPersistentStorage> GetPersistentStorageAsync(
+            this Solution solution, bool checkBranchId, CancellationToken cancellationToken)
+        {
+            return GetPersistentStorageAsync(solution.Workspace.Services, solution.Options, SolutionKey.ToSolutionKey(solution), checkBranchId, cancellationToken);
+        }
+
+        public static ValueTask<IChecksummedPersistentStorage> GetPersistentStorageAsync(
+            this HostWorkspaceServices services, OptionSet options, SolutionKey solutionKey, bool checkBranchId, CancellationToken cancellationToken)
+        {
+            return GetPersistentStorageAsync(services, GetPersistentStorageDatabase(options), solutionKey, checkBranchId, cancellationToken);
+        }
+
+        public static ValueTask<IChecksummedPersistentStorage> GetPersistentStorageAsync(
+            this HostWorkspaceServices services, StorageDatabase database, SolutionKey solutionKey, bool checkBranchId, CancellationToken cancellationToken)
+        {
+            var persistenceStorageService = services.GetPersistentStorageService(database);
+
+            var configuration = services.GetRequiredService<IPersistentStorageConfiguration>();
+            return persistenceStorageService.GetStorageAsync(configuration, solutionKey, checkBranchId, cancellationToken);
+        }
```

Comment:
helpers to make consumption fo the persistence service easier.  Almost all consumers should just call one of these.